riscv/processor.cc

```diff
@@ -533,6 +533,16 @@ void state_t::reset(processor_t* const proc, reg_t max_isa)
   if (proc->extension_enabled(EXT_ZCMT))
     csrmap[CSR_JVT] = jvt = std::make_shared<jvt_csr_t>(proc, CSR_JVT, 0);
 
+  if (proc->extension_enabled(EXT_ZICFISS)) {
+    csrmap[CSR_SSP] = ssp = std::make_shared<basic_csr_t>(proc, CSR_SSP, 0);
```

Comment:
Add a write mask for ssp so that bits 1:0 are read-only-zero.
Spike does not support changing XLEN. So if if proc->get_const_xlen is != 32 then also include bit 2 to the mask so bit 2 is also read-only-zero.

Reply:
Access to ssp CSR requires SSE in corresponding envcfg. This is missing.
> Attempts to access the ssp CSR may result in either an illegal-instruction exception or a virtual
instruction exception, contingent upon the state of the xenvcfg.SSE fields. The conditions are
specified as follows:
> • If the privilege mode is less than M and menvcfg.SSE is 0, an illegal-instruction exception is
raised.
> • Otherwise, if in U-mode and senvcfg.SSE is 0, an illegal-instruction exception is raised.
> • Otherwise, if in VS-mode and henvcfg.SSE is 0, a virtual instruction exception is raised.
> • Otherwise, if in VU-mode and either henvcfg.SSE or senvcfg.SSE is 0, a virtual instruction exception is raised.
> • Otherwise, the access is allowed.

Reply:
Maybe I should create new csr type with special permission
```
void ssp_csr_t::verify_permissions(insn_t insn, bool write) const {
  DECLARE_XENVCFG_VARS(SSE);
  require_envcfg(SSE);
}
```